tests/test_unknown_platforms.py

```diff
@@ -72,3 +88,174 @@ def test_get_sensor_given_type(self):
+        def can_load_this_filename(self, filename):
+            return True
+
+        def can_load_this_type(self, suffix):
+            return True
+
+        def can_load_this_file(self, file_contents):
+            return True
+
+        def _load_this_file(self, data_store, path, file_object, datafile, change_id):
+            self.test_platform = self.get_cached_platform(
+                data_store, None, change_id=change_id, unknown=True
+            )
+
+    ds = DataStore("", "", "", 0, ":memory:", db_type="sqlite")
+    ds.initialise()
+
+    with ds.session_scope():
+        ds.populate_reference()
+        ds.populate_metadata()
+
+    test_parser = TestParser()
+
+    processor = FileProcessor()
+    processor.register_importer(test_parser)
+
+    with ds.session_scope():
+        processor.process(SINGLE_REP_FILE, ds, False)
+        ds.session.add(test_parser.test_platform)
+        _ = uuid.UUID(test_parser.test_platform.name)
+        assert test_parser.test_platform.platform_type_name == "Unknown"
+        assert test_parser.test_platform.nationality_name == "Unknown"
+
+
+class UnknownPlatformsResolverTestCase(unittest.TestCase):
+    def setUp(self):
+        self.ds = DataStore(
+            "", "", "", 0, ":memory:", db_type="sqlite", missing_data_resolver=CommandLineResolver()
+        )
+        self.ds.initialise()
+
+        with self.ds.session_scope():
+            self.ds.populate_reference()
+
+        self.processor = FileProcessor()
+        self.processor.load_importers_dynamically()
+
+    @patch("pepys_import.file.file_processor.prompt")
+    @patch("pepys_import.resolvers.command_line_input.prompt")
+    @patch("pepys_import.resolvers.command_line_resolver.prompt")
+    def test_answering_store_as_unknown(self, prompt, menu_prompt, processor_prompt):
+        menu_prompt.side_effect = [
+            "2",  # Public
+            "1",  # Yes, correct
+            "3",  # Store platform as Unknown
+        ]
+
+        processor_prompt.side_effect = ["2"]
+
+        self.processor.process(SINGLE_NISIDA_FILE, data_store=self.ds)
+
+        with self.ds.session_scope():
+            # Should have imported the right number of State entries
+            states = self.ds.session.query(self.ds.db_classes.State).all()
+            assert len(states) == 5
+
+            # The platform that has been imported should be marked as Unknown
+            platforms = self.ds.session.query(self.ds.db_classes.Platform).all()
+            assert len(platforms) == 1
+
+            plat = platforms[0]
+
+            assert plat.name == "ADRI"
+            assert plat.platform_type_name == "Unknown"
+            assert plat.nationality_name == "Unknown"
+
+    @patch("pepys_import.file.file_processor.prompt")
+    @patch("pepys_import.resolvers.command_line_input.prompt")
+    @patch("pepys_import.resolvers.command_line_resolver.prompt")
+    def test_answering_store_remaining_as_unknown(self, prompt, menu_prompt, processor_prompt):
```

Comment:
the linter is saying prompt is unused. It is unused, but is it still a necessary import?

Reply:
I was just writing a message to say that it was needed and it was a spurious warning - but then realised that it could be removed, so I've removed it.